src/core/agent/agent.ts

```diff
@@ -264,9 +265,12 @@ class Agent {
     let bran = "";
     try {
       const mnemonic = seedPhrase.join(" ");
-      bran = Buffer.from(mnemonicToEntropy(mnemonic), "hex")
-        .toString("utf-8")
-        .replace(/\0/g, "");
+      const entropy = mnemonicToEntropy(mnemonic);
+      let branBuffer = Buffer.from(entropy, "hex");
+      branBuffer = branBuffer.slice(0, -Agent.BUFFER_ALLOC_SIZE);
+      const hexString = branBuffer.toString("hex");
+
+      bran = Buffer.from(hexString, "hex").toString("utf-8");
```

Comment:
```
const branBuffer = branBuffer = Buffer.from(entropy, "hex").slice(0, -Agent.BUFFER_ALLOC_SIZE);
bran = branBuffer.toString("utf-8");
```
does this work? it would avoid non const variables and there seems to be an unnecessary conversion from hex and back in the middle

Reply:
yess, it work